### ISSUE TITLE
RuntimeError: unauthorized use of script (1:10624) after adding comment to outer key

### ISSUE THREAD

**Issue by goraje** [dashingsoft/pyarmor#1581]
```
INFO     Python 3.9.16
INFO     Pyarmor 8.4.1 (group), 006005, ******
INFO     Platform darwin.aarch64
INFO     search inputs ...
INFO     find package at src/foo
```
I have added a comment plugin as described in https://pyarmor.readthedocs.io/en/stable/tutorial/customization.html#comments-within-outer-key to include the expiration date in the outer key. I have created `comment_plugin.py` in `.pyarmor` and executed:
```
pyarmor cfg plugins + "comment_plugin"
```
The contents of the plugin is as follows:
```
from datetime import datetime

__all__ = ['CommentPlugin']

class CommentPlugin:

    @staticmethod
    def post_key(ctx, keyfile, **keyinfo):
        expired = None
        for name, value in keyinfo.items():
            print(name, value)
            if name == 'expired':
               expired = datetime.strptime(value, ".%Y-%m-%d")

        if expired:
            print('patching runtime key')
            comment = '# expired date: %s\n' % expired
            with open(keyfile, 'rb') as f:
                keydata = f.read()
            with open(keyfile, 'wb') as f:
                f.write(comment.encode())
                f.write(keydata)
```
I've made changes to read the expiration date properly for my use case since I'm running:
```
pyarmor gen key -O "keylist" -e .2023-11-20
INFO     Python 3.9.16
INFO     Pyarmor 8.4.1 (group), 006005, ******
INFO     Platform darwin.aarch64
INFO     start to generate outer runtime key "pyarmor.rkey"
INFO     write keylist/pyarmor.rkey
expired .2023-11-03
devices None
period None
data None
patching runtime key
INFO     generate outer runtime key OK
```
After I try to run the Python package that uses this outer runtime key I get (`foo_run` is an entrypoint for my `foo` package):
```
foo_run
Traceback (most recent call last):
  File "/Users/jgora/pyarmor_venv3.9/bin/foo_run", line 5, in <module>
    from foo import main
  File "/Users/jgora/pyarmor_venv3.9/lib/python3.9/site-packages/foo/__init__.py", line 1, in <module>
    from . import main
  File "/Users/jgora/pyarmor_venv3.9/lib/python3.9/site-packages/foo/main.py", line 7, in <module>
    from .pyarmor_runtime_006005.pyarmor_runtime import __pyarmor__
  File "/Users/jgora/pyarmor_venv3.9/lib/python3.9/site-packages/foo/pyarmor_runtime_006005/__init__.py", line 2, in <module>
    from .pyarmor_runtime import __pyarmor__
RuntimeError: unauthorized use of script (1:10624)
```
This does not happen if the outer runtime key has no comments in it.
It also does not happen if the plugin makes no changes to the outer key content:
```
from datetime import datetime

__all__ = ['CommentPlugin']

class CommentPlugin:

    @staticmethod
    def post_key(ctx, keyfile, **keyinfo):
        with open(keyfile, 'rb') as f:
            keydata = f.read()
        with open(keyfile, 'wb') as f:
            f.write(keydata)
```
I just get the expected output that looks as follows:
```
foo_run
The package is protected.
The code expires at: 20 Nov 2023
THIS IS A TEST
```
## Build environment for issue replication

- requires the Python `wheel` package to be installed
- requires `rsync`
- run `make` to execute code obfuscation and build a wheel package of `foo` in `dist` (the path for the copy command in the Makefile where I move the runtime key needs to be adjusted because it depends on the license number `pyarmor_runtime_006005` in our case)

```
├── .pyarmor
│   ├── comment_plugin.py
│   ├── config
│   └── pyarmor.debug.log
├── MANIFEST.in
├── Makefile
├── setup.py
└── src
    └── foo
        ├── __init__.py
        └── main.py
```

- `.pyarmor/comment_plugin.py`
```
from datetime import datetime

__all__ = ['CommentPlugin']

class CommentPlugin:

    @staticmethod
    def post_key(ctx, keyfile, **keyinfo):
        expired = None
        for name, value in keyinfo.items():
            print(name, value)
            if name == 'expired':
               expired = datetime.strptime(value, ".%Y-%m-%d")

        if expired:
            print('patching runtime key')
            comment = '# expired date: %s\n' % expired
            with open(keyfile, 'rb') as f:
                keydata = f.read()
            with open(keyfile, 'wb') as f:
                f.write(comment.encode())
                f.write(keydata)
```
- `.pyarmor/config`
```
[finder]
excludes = *__pycache__ src/foo/__init__.py src/foo/main.py

[builder]
plugins = CodesignPlugin DarwinUniversalPlugin comment_plugin
```
- `MANIFEST.in`
```
recursive-include src/foo/pyarmor_runtime_006005 *
```
- `Makefile`
```
PACKAGE_NAME := $(shell find src/* -type d -print -maxdepth 0 | xargs -I {} basename {})
PACKAGE_DIR := "src"
PACKAGE_BKP := "bkp"
ARMORED_DIR := "armored"
KEYLIST_DIR := "keylist"

default:
	@echo "Package:" $(PACKAGE_NAME)
	@echo "Running pyarmor"
	pyarmor gen --outer --obf-code 2 --enable-jit -O $(ARMORED_DIR) -i $(PACKAGE_DIR)/$(PACKAGE_NAME)
	pyarmor gen key -O $(KEYLIST_DIR) -e .2023-11-20
	@echo "Syncing src"
	cp -r $(PACKAGE_DIR) $(PACKAGE_BKP)
	rsync -avh $(ARMORED_DIR)/ $(PACKAGE_DIR)
	cp $(KEYLIST_DIR)/pyarmor.rkey $(PACKAGE_DIR)/$(PACKAGE_NAME)/pyarmor_runtime_006005
	@echo "Build wheel"
	python setup.py bdist_wheel
	@echo "Cleanup"
	rm -rf build $(ARMORED_DIR) $(PACKAGE_DIR) $(KEYLIST_DIR)
	mv $(PACKAGE_BKP) $(PACKAGE_DIR)
```
- `setup.py`
```
#!/usr/bin/env python

from __future__ import annotations


from setuptools import find_packages, setup


setup(
    name="foo",
    description="Foo package",
    author="John Doe",
    author_email="jondoe@jondoe.com",
    version="0.1.0",
    license_files=["LICENSE.md"],
    packages=find_packages(where="src"),
    package_dir={"": "src"},
    include_package_data=True,
    python_requires=">=3.9",
    install_requires=[],
    entry_points={
        "console_scripts": {
            "foo_run = foo:main"
        }
    }
)
```
- `src/foo/__init__.py`
```
from . import main


def main():
    print("THIS IS A TEST")
```
- `src/foo/main.py`
```
from datetime import datetime

class LicenseError(Exception):
    pass

try:
    from .pyarmor_runtime_006005.pyarmor_runtime import __pyarmor__

    print("The package is protected.")
    expiration_date = datetime.fromtimestamp(__pyarmor__(1, None, b'keyinfo', 1))
    print(expiration_date.strftime("The code expires at: %d %b %Y"))

except RuntimeError as exc:
    if "license key is expired " in str(exc):
        raise LicenseError("MY ERROR")
    else:
        raise

except ModuleNotFoundError:
    print("The package is not protected.")
```

**Comment by jondy** [comment]
Reproduced, it will be fixed in later release

A suggestion, for this bug, using a simple hello-world script is enought as example, it could save your time to write issue and my time to read issue, :)

**Comment by jondy** [comment]
Verify it by downloading this temporary extension
https://pyarmor.dashingsoft.com/downloads/temp/5.4.2/darwin.arm64/libs/cp3.9/pyarmor_runtime.so

And overwrite `/path/to/pyarmor/cli/core/pyarmor_runtime.so`

**Comment by goraje** [comment]
Hi, replacing `pyarmor_runtime.so` in `.../cli/core` with the one provided by you did indeed fix the error that I was getting. Thank you for your help :)

**Comment by jondy** [comment]
Fixed in v8.4.4
